### DISCUSSION TITLE
SubscribeForNotifications works only for one call.  Realm 11.6.0

### DISCUSSION THREAD

**Discussion by alpitaR** [realm/realm-dotnet#3610] [Q&A]
I am using SubscribeForNotifications  on some Query to update list view ItemSource when realm object changes 

```csharp
 var stuChanged = realmDB.All<Student>().SubscribeForNotifications((sender, changes) =>
 {
     if (changes != null)
         UpdateFromDB();
 });
```

When i add a new student `UpdatefromDB()` gets called . However it does not gets triggered if i add more students. Below is how i am adding a student to realm . 

```csharp
 realmDB.Write(() => { realmDB.Add(student); });
 using (var trans = realmDB.BeginWrite())
 {
     realmDB.Add(student);
     if (trans.State == TransactionState.Running)
     {
         trans.Commit();
     }
 }
```
Any thoughts around this?

**Comment by nirinchev** [comment]
My guess is that you need to assign `realmDB.All<Student>()` to a local variable as well - otherwise the collection you're listening to gets garbage collected. This is due to https://github.com/realm/realm-dotnet/issues/2837.
